keras/engine/topology.py

```diff
@@ -413,13 +413,16 @@ def assert_input_compatibility(self, inputs):
             ValueError: in case of mismatch between
                 the provided inputs and the expectations of the layer.
         """
+        inputs = _to_list(inputs)
+        for x in inputs:
+            K.is_keras_tensor(x)
```

Comment:
We should except `ValueError` then raise a `ValueError` with a more precise error message